Incorrect management of some addresses (about 5%)

csauge opened the pull request:
- Address issue #12
- Move the all parsing of addresses to readJson.py:
  - This can be reusable to all other billetteries (new issue to open)
  - Thus we can remove the former get_address_data and strip_zip_code
- Enhance logic for address:
  - 40 events are newly added (they were ignored until now)
  - 10 incorrect events are removed (mainly international events)
  - 10 differents addresses are more accurate
  - Remove the usage of str.title() which lead to incorrect cases

thomas-bouvier commented on utils/readJson.py:
```diff
@@ -1,6 +1,81 @@
+    words = full_address.split()
+
+    # create a list of address removing one word for each line until three words minimum
+    if lines := "\n".join(' '.join(words[i:]) for i in range(len(words)) if len(words[i:]) >= 3):
+        url = "https://api-adresse.data.gouv.fr/search/csv/"
+        files = {'data': StringIO("query\n" + lines)}
+        response = requests.post(url, files=files)
+        if data_list := list(csv.DictReader(StringIO(response.text))):
+            data_list = list(map(lambda x: {**x, 'result_score': float(x['result_score']) if x['result_score'] != '' else 0.0}, data_list))
+            max_data = max(data_list, key=lambda x:x['result_score'])
+            if max_data["result_score"] > 0.5:  # keep only good results
```
I feel that this part should be explained thoroughly, especially given that what we're trying to achieve here is a bit hacky. Commenting the function will do.

thomas-bouvier commented:
@csauge, can you please run the [compare.py](https://github.com/trouver-une-fresque/trouver-une-fresque/blob/0d84d159551d602e2f3338f766021d967f36c497/compare.py) script with the outputs of the scraper without and with this contribution? You've probably already done it, but I think it would be a good habit to post the results here :)

Also, comments on the function `get_default_address` would be appreciated. Does this method produce relevant results?

csauge commented:
You are right. I have: 
- added a new commit to comment the tricky address part
- added in this comments:
    - compare.txt from your compare.py but this is not really accurate as some bad events (international or truncated locations) are removed and other are added
    - list of full_locations before my fix (results1_full_location.txt) and after my fix (results2_full_location.txt) for billetweb only
    - list of json before my fix (events_20240208_114802.json) and after my fix (events_20240208_113552.json) for billetweb only

[compare.txt](https://github.com/trouver-une-fresque/trouver-une-fresque/files/14208404/compare.txt)
[results1_full_location.txt](https://github.com/trouver-une-fresque/trouver-une-fresque/files/14208406/results1_full_location.txt)
[results2_full_location.txt](https://github.com/trouver-une-fresque/trouver-une-fresque/files/14208407/results2_full_location.txt)
[events_20240208_114802.json](https://github.com/trouver-une-fresque/trouver-une-fresque/files/14208411/events_20240208_114802.json)
[events_20240208_113552.json](https://github.com/trouver-une-fresque/trouver-une-fresque/files/14208412/events_20240208_113552.json)

thomas-bouvier commented:
Thank you very much @csauge.

I have a question: why are some words now spelled out in full? Like `20 All. de la Maison Rouge` -> `20 Allée de la Maison Rouge`.

On a side note, some street numbers are now missing, like `101 Rue Soffrey de Calignon` -> `Rue Soffrey de Calignon`. But the improvements far outweigh this detail.

csauge commented:
Thank you Thomas.
Indeed, some street numbers are missing because of address.gouv.fr which does not recognize some street numbers (strange, but maybe it will be fixed in the future or by using another API)... But this is not a new problem because before you had (in previous version of code) the same results for longitude and latitude, this is just you stored in "address" json the query value and not what address.gouv.fr returned you as address. I think this is more coherent to store, as with this commit, the returned address.gouv.fr "address" to be inline with the longitude and latitude.

thomas-bouvier commented:
Thanks, I missed this
